pkg/validationfile/loader.go

```diff
@@ -170,3 +183,42 @@ func PopulateFromFilesContents(ctx context.Context, ds datastore.Datastore, cave
 
 	return &PopulatedValidationFile{schemaStr, objectDefs, caveatDefs, rels, files}, revision, err
 }
+
+// CompileSchema takes a SchemaWithPosition and returns the compiled schema, or else an error.
+// TODO: this is probably the wrong place for this, in part because it's coupling to the compiler
+// implementation.
+func CompileSchema(schemaWithPosition blocks.SchemaWithPosition) (*compiler.CompiledSchema, error) {
```

Comment:
Have this take in the caveat type set as well, and pass it via option to the compile call below